pkg/services/thumbs/repo.go

```diff
@@ -0,0 +1,77 @@
+package thumbs
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func newThumbnailRepo(store *sqlstore.SQLStore) thumbnailRepo {
+	repo := &sqlThumbnailRepository{
+		store: store,
+	}
+	return repo
+}
+
+type sqlThumbnailRepository struct {
+	store *sqlstore.SQLStore
+}
+
+func (r *sqlThumbnailRepository) saveFromFile(filePath string, meta models.DashboardThumbnailMeta, dashboardVersion int) (int64, error) {
+	content, err := os.ReadFile(filepath.Clean(filePath))
```

Comment:
I think that it's safer to make sure that `filepath` is rooted, given rule 4 of [filepath.Clean()](https://pkg.go.dev/path/filepath#Clean) if the provided path is relative (i.e. not rooted) any occurrences of .. will remain in the resulting path.

Reply:
added a comment explaining the source of the `filepath` variable

Reply:
also added `isAbs()` validation